iOS/Collection View Management /CollectionViewDataSource.swift

```diff
@@ -11,6 +11,7 @@ import Foundation
 /// A base class for types that can act as a data source for a UICollectionview.
 /// Subclasses should override functions related to dequeuing cells and supplementary views.
 /// This class works the same as UICollectionViewDiffableDataSource but it allows you to subclass it more easily and hold additional state.
+@MainActor
```

Comment:
Datasource changes need to be done only on the main actor or only off the main actor. Not both. Making this a main actor only class helps avoid those issues.